support/infra/src/storage.rs

```diff
@@ -1,32 +1,67 @@
-use crate::types::{FileChunk, FileKey, FileMetadata};
+use sp_core::H256;
+use sp_trie::CompactProof;
+
+use crate::types::{Chunk, Key, Metadata};
+
+/// Leaf in the Forest or File trie.
+pub struct Leaf<D> {
+    pub key: Key,
+    pub data: D,
+}
+
+/// Proving either the exact key or the neighbour keys of the challenged key.
+pub enum Proven<D> {
+    ExactKey(Leaf<D>),
+    NeighbourKeys((Leaf<D>, Leaf<D>)),
+}
+
+pub struct FileProof {
+    /// The file key that was proven.
+    pub proven: Proven<Chunk>,
+    /// The compact proof.
+    pub proof: CompactProof,
+    /// The root hash of the trie.
+    pub root_hash: Key,
+}
 
 /// Storage interface to be implemented by the storage providers.
-pub trait Storage: Clone + Send + Sync + 'static {
+pub trait FileStorage: 'static {
```

Comment:
Why are all these structs and and traits defined here and not in forest-manager or wherever else they're used? Do they need to be shared with others somehow?

Reply:
Initially I added a mock Storage trait here and went from there. It makes more sense for each to be next to the implementations as well.